Jenkinsfile.release

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2017-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@Library(['private-pipeline-library', 'jenkins-shared']) _
+
+import com.sonatype.jenkins.shared.Expectation
+
+properties([
+  parameters([
+    run(name: 'releaseBuild',
+        filter: 'SUCCESSFUL',
+        projectName: 'insight/insight-brain/release',
+        description: 'The latest release of IQ Server to build the docker image')
+  ])
+])
+
+String imageName = 'sonatype/nexus-iq-server',
+String dockerHubRepository = 'nexus-iq-server',
+String version = ''
+String checksum = ''
+
+dockerizedBuildPipeline(
+  deployBranch: 'main',
+  prepare: {
+    githubStatusUpdate('pending')
+    version = getVersionFromBuildName(env.releaseBuild_NAME)
+    checksum = readBuildArtifact('insight/insight-brain/release', env.releaseBuild_NUMBER, "artifacts/nexus-iq-server-${version}-bundle.tar.gz.sha256").trim()
+    updateIQServerVersionAndChecksum(version, checksum)
+    commitAndPushChanges(version)
+  },
```

Comment:
Is this order different to previously?

IIUC previously we were committing and pushing the version updates after building/testing/evaluating but here it looks like we're committing and pushing the version updates before building/testing/evaluating?

One reason this might matter is in the past we've had to re-run the release because of evaluation failures
https://jenkins.ci.sonatype.dev/job/insight/job/insight-brain/job/docker/job/docker-nexus-iq-server/
so if we commit/push before we might end up with some unnecessary commits

On the other hand this would help avoid the issue of commits to main whilst a release is happening.

Not sure if we want to change the process as part of this though.

Reply:
You're right, this version doesn't follow the exact order of steps. I can do this step in the `postDeploy` stage if appropriate.

Reply:
> You're right, this version doesn't follow the exact order of steps. I can do this step in the `postDeploy` stage if appropriate.

I would avoid changing this as part of this PR and try to default to how it was working previously if possible. Changing the order may be something we have to discuss in a larger context

Reply:
> The following [code](https://github.com/sonatype/docker-nexus-iq-server/blob/b1722861de1577da808fe846dcdbd6dc8dfb8cdd/Jenkinsfile.release#L105)  runSafely "git diff --exit-code --cached || git commit -m 'Update IQ version to ${version}.'" will only commit if an actual code change is detected. So if the same release number is used, no change would be detected and a commit shouldn't happen.

Nice spot, so if we specify the exact same IQ Server version, it won't make a commit.

Also if the build/test/evaluation fails it won't have pushed the image so we shouldn't have a problem there either.

The tagging should be good too, it may not always point to the "Update IQ Server to X." commit (if the release is reran but someone committed to main before the rerun) but I think that's fine.

I agree that we should be ok (no changes needed @agustin-mrtz ) and can give this a try.

Thanks @shaikhu 👍